secret/secrets_manager_client_test.go

```diff
@@ -40,6 +40,17 @@ func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
 	})
 	require.NoError(t, err)
 
+	Cleanup := func(out *secretsmanager.CreateSecretOutput) {
+		if out != nil && out.Name != nil && out.ARN != nil {
+			out, err := c.DeleteSecret(ctx, &secretsmanager.DeleteSecretInput{
+				ForceDeleteWithoutRecovery: aws.Bool(true),
+				SecretId:                   out.ARN,
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+		}
+	}
+
 	t.Run("CreateFailsWithInvalidInput", func(t *testing.T) {
```

Comment:
I would prefer that these tests be re-written in test matrix style rather than using `t.Run` for each test case. See [this example](https://github.com/mongodb/jasper/blob/4a44d3330b808c341efea44ef6e88cea20c922ad/tracker_windows_test.go#L32-L129) (ping me if you need more explanation).

The benefit of doing this is that:
* You can pass an individual context with timeouts to each test (like [in the example](https://github.com/mongodb/jasper/blob/4a44d3330b808c341efea44ef6e88cea20c922ad/tracker_windows_test.go#L116)), so you can say that in total, the tests run for however long, but each test shouldn't take more than maybe 30 seconds or so (which seems like an acceptable timeout for tests that require integration with AWS).
* You can do common setup easily and pass it into each test (so for example, all the tests depend on first having a secrets manager client).
* Tests are run in random order, since Go maps don't guarantee any order on the keys/values returned from it. This will expose problems such as one test depending on the result of another test. Tests should be isolated from one another and should not depend on the order that they're run.

Reply:
The example code has a defer that cleans up the tracker, so should I also write a defer to cleanup the client with its `Close` function?

Reply:
Yeah, I would call Close just to be on the safe. side For now, it's not necessary (because `Close` won't do anything), but we might have it do more resource cleanup in the future and it doesn't hurt to call it.